monkey/agent_plugins/exploiters/wmi/src/wmi_client.py

```diff
@@ -30,46 +34,57 @@ class WMIClient:
     def __init__(self):
         self._wbem_services: Optional[wmi.IWbemServices] = None
         self._connected = False
+        self._dcom: Optional[DCOMConnection] = None
+        DCOMConnection.PINGTIMER = Timer(1, DCOMConnection.pingServer)
 
     def connected(self) -> bool:
         return self._connected
 
-    def login(self, host: TargetHost, credentials: Credentials):
+    def login(self, host: TargetHost, credentials: Credentials, timeout: float):
         """
         Login to the remote host
 
         :param host: Remote host
         :param credentials: Credentials to use
         :raises Exception: If login fails
         """
-        # Impacket has a hard-coded timeout of 30 seconds
-        dcom = DCOMConnection(
-            str(host.ip),
-            username=credentials.identity.username,
-            password=get_plaintext(secret_of_type(credentials, Password)),
-            domain=str(host.ip),
-            lmhash=get_plaintext(secret_of_type(credentials, LMHash)),
-            nthash=get_plaintext(secret_of_type(credentials, NTHash)),
-            oxidResolver=True,
-        )
-
+        # Impacket has a hard-coded timeout of 120 seconds
+        # so we set our own preffered one
+        # DCOMConnection.PINGTIMER = Timer(timeout, DCOMConnection.pingServer)
```

Comment:
We may just need to live with this. #3557 will prevent the agent from hanging during shutdown. We can create an issue and pursue a resolution at a later date if it's a priority. I think the only real effect, here, is that the WMI exploiter will cause propagation to be very slow in some environments.